### PR TITLE
WELD-2465 Switch to JBoss Commons Annotation 1.3 version.

### PR DESCRIPTION
**Pull request by manovotn**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -62,6 +62,7 @@
         <jandex.version>2.0.3.Final</jandex.version>
         <javax.activation.version>1.1</javax.activation.version>
         <javax.el.version>3.0.0</javax.el.version>
+        <jboss.spec.annotations-api.version>1.0.0.Final</jboss.spec.annotations-api.version>
```
Comment (mkouba) [inline pom.xml]: This should be in Weld API pom: https://github.com/weld/api/blob/master/pom.xml#L108-L110

Comment (manovotn) [reply pom.xml]: Ah, good point, thanks

Comment (manovotn) [reply pom.xml]: Hmm, it seems like we had this dep duplicated in core and API...supposedly to avoid having to release API?

**Comment by manovotn** [comment]
Requires API upgrade - see https://github.com/weld/api/pull/75

**Comment by manovotn** [comment]
@mkouba this should be ready for merging.

**Pull request merged**